g4me/analysis/PrepareTracksForMatchingAndFit_embedding.C

```diff
@@ -0,0 +1,330 @@
+	       new (trackCandidatesHitCovITS[nPreparedTracksITS]) TClonesArray(allTracksHitCovITS.at(iTrack));
+	       idTrackITS.emplace_back(iTrack);
+	       TParticle part;
+	       part.SetPdgCode(io_underlaying.tracks.pdg[iTrack]);
+	       part.SetProductionVertex(io_underlaying.tracks.vx[iTrack],io_underlaying.tracks.vy[iTrack],io_underlaying.tracks.vz[iTrack],io_underlaying.tracks.vt[iTrack]);
+	       part.SetMomentum(io_underlaying.tracks.px[iTrack],io_underlaying.tracks.py[iTrack],io_underlaying.tracks.pz[iTrack],0.001*io_underlaying.tracks.e[iTrack]);
+         new (particlesITS[nPreparedTracksITS]) TParticle(part);
+         nPreparedTracksITS++;
+      }
+    }
+
+    // filling the final arrays with the hit information from selected MID tracklets
+
+    nPreparedTrackletsMID = 0;
+    TVector3 posHitMID1, posHitMID2;
+    int idHitLayer1, idHitLayer2, trackIdHitLayer1, trackIdHitLayer2, trackletID;
+
+    for (int iHitLayer1=0; iHitLayer1<nHits_MIDLayer1; iHitLayer1++) {
+
+      idHitLayer1 = arrayHitID_MIDLayer1[iHitLayer1];
+      posHitMID1.SetXYZ(gRandom->Gaus(io_underlaying.hits.x[idHitLayer1],resolutionMID),gRandom->Gaus(io_underlaying.hits.y[idHitLayer1],resolutionMID),gRandom->Gaus(io_underlaying.hits.z[idHitLayer1],resolutionMID));
+      trackIdHitLayer1 = io_underlaying.hits.trkid[idHitLayer1];
+
+      for (int iHitLayer2=0; iHitLayer2<nHits_MIDLayer2; iHitLayer2++) {
+
+	       idHitLayer2 = arrayHitID_MIDLayer2[iHitLayer2];
+         posHitMID2.SetXYZ(gRandom->Gaus(io_underlaying.hits.x[idHitLayer2],resolutionMID),gRandom->Gaus(io_underlaying.hits.y[idHitLayer2],resolutionMID),gRandom->Gaus(io_underlaying.hits.z[idHitLayer2],resolutionMID));
+	       trackIdHitLayer2 = io_underlaying.hits.trkid[idHitLayer2];
+
+	       if (trackletSel->IsMIDTrackletSelected(posHitMID1,posHitMID2,kFALSE)) {
+
+	          if (trackIdHitLayer1==trackIdHitLayer2) trackletID = trackIdHitLayer1;
+	          else                                    trackletID = -1;
+
+	          TClonesArray trackletMIDpos("TVector3");
+	          TClonesArray trackletMIDcov("TMatrixDSym");
+
+	          new (trackletMIDpos[trackletMIDpos.GetEntries()]) TVector3(posHitMID1);
+	          new (trackletMIDpos[trackletMIDpos.GetEntries()]) TVector3(posHitMID2);
+	          new (trackletMIDcov[trackletMIDcov.GetEntries()]) TMatrixDSym(covMID);
+	          new (trackletMIDcov[trackletMIDcov.GetEntries()]) TMatrixDSym(covMID);
+
+	          new (trackCandidatesHitPosMID[nPreparedTrackletsMID]) TClonesArray(trackletMIDpos);
+	          new (trackCandidatesHitCovMID[nPreparedTrackletsMID]) TClonesArray(trackletMIDcov);
+
+	          idTrackMID.emplace_back(trackletID);
+
+	          nPreparedTrackletsMID++;
+
+	      }
+      }
+    }
+
+    // end of loop over pythia tree
+    //--------------------------------------------------------------------------
+    // Loop over muon hits
+    for (int iHit=0; iHit<io_signal.hits.n; iHit++) {
```

Comment:
In the current version of the code, MID tracklets are formed combining pairs of MID hits coming both from the underlying event or both from the signal event. So nHits_MIDLayer1 and nHits_MIDLayer2 should be reset to zero when reading the signal event...